### ISSUE TITLE
Method with an enum argument and default value can't be mocked

### ISSUE THREAD

**Issue by bicpi** [phpspec/prophecy#567]
Hi,
given this enum (PHP 8.1+):

```php
enum AppType
{
    case IOS;
    case ANDROID;
}
```

And an interface with a method using the enum as default argument of a method:

```php
interface BarService
{
    public function doIt(AppType $appType = AppType::ANDROID): void;
}
```

And a service using the interface as a dependency:

```php
class FooService
{
    public function __construct(
        readonly private BarService $bar
    ) {
    }

    public function execute(): void
    {
        $this->bar->doIt();
    }
}
```

The following test fails:

```php
namespace Tests;

use PHPUnit\Framework\TestCase;
use Prophecy\PhpUnit\ProphecyTrait;

enum AppType {...}
interface BarService {...}
class FooService {...}

class FooServiceTest extends TestCase
{
    use ProphecyTrait;

    /**
     * @test
     */
    public function it_fails(): void
    {
        $someService = $this->prophesize(BarService::class);
        $someService
            ->doIt()
            ->shouldBeCalled();

        $fooService = new FooService($someService->reveal());
        $fooService->execute();
    }
}
```

The error message is:

```
Error : Class "Double\BarService\Tests\AppType" not found
```

When passing the default value explicitly works though.

**Comment by stof** [comment]
This error has nothing to do with `new` in initializers (there is no `new` in your initializers). It is about support for enums in default values of an argument.

**Comment by bicpi** [comment]
True, I think I also observed an issue with `new` usage, I'll refine the description.

**Comment by stof** [comment]
If there is also an issue with `new` (which is likely as we haven't done anything to support it), please open a separate issue for it (as the fix might not be the same).

**Comment by bicpi** [comment]
I reduced this issue to the problem with the enum argument; after testing the case with the `new` usage in a method's default value, I created the separate issue https://github.com/phpspec/prophecy/issues/568.

**Comment by jdreesen** [comment]
I can also confirm this. The problem is, that the generated default value is in the wrong namespace, because the leading `\` is missing. 

This seems only to be a problem in PHP 8.1, see: https://3v4l.org/3JPCM

I proposed a fix: #618
